File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/EllipseShape.java
package com.hello.uxml.tools.framework;

import com.hello.uxml.tools.framework.graphics.UISurface;

/**
 * Defines an elliptical shape.
 *
 * @author ferhat@
 *
 */
public class EllipseShape extends Shape {

  @Override
  protected void onMeasure(double availableWidth, double availableHeight) {
    setMeasuredDimension(getWidth(), getHeight());
  }

  @Override
  protected void onRedraw(UISurface surface) {
    surface.drawEllipse(getFill(), getStroke(), new Rectangle(0, 0, getWidth(), getHeight()));
  }
}

File: framework/lib/ellipse_shape.dart
part of uxml;

/**
 * Defines an elliptical shape.
 *
 * @author ferhat@ (Ferhat Buyukkokten)
 */
class EllipseShape extends Shape {

  static ElementDef ellipseElementDef;

  EllipseShape() : super() {
  }

  /** Overrides UIElement.onRedraw. */
  void onRedraw(UISurface surface) {
    surface.clear();
    if (fill != null || stroke != null) {
      surface.drawEllipse(0.0, 0.0, layoutWidth, layoutHeight, fill, stroke);
    }
  }

  /** Registers component. */
  static void registerEllipseShape() {
    ellipseElementDef = ElementRegistry.register("EllipseShape",
        Shape.shapeElementDef, null, null);
  }

  /** @see UxmlElement.getDefinition. */
  ElementDef getDefinition() => ellipseElementDef;
}
